mapviz_plugins/src/point_drawing_plugin.cpp

```diff
@@ -0,0 +1,485 @@
+  {
+    bool transformed = false;
+
+    std::list<StampedPoint>::iterator points_it = points_.begin();
+    for (; points_it != points_.end(); ++points_it)
+    {
+      transformed = transformed | TransformPoint(*points_it);
+    }
+
+    transformed = transformed | TransformPoint(cur_point_);
+    if (laps_.size() > 0)
+    {
+      for (int i = 0; i < laps_.size(); i++)
+      {
+        std::list<StampedPoint>::iterator lap_it = laps_[i].begin();
+        for (; lap_it != laps_[i].end(); ++lap_it)
+        {
+          transformed = transformed | TransformPoint(*lap_it);
+        }
+      }
+    }
+    if (!points_.empty() && !transformed)
+    {
+      PrintError("No transform between " + cur_point_.source_frame + " and " +
+                 target_frame_);
+    }
+  }
+
+  bool PointDrawingPlugin::DrawLaps()
+  {
+    bool transformed = false;
+    glColor4f(color_.redF(), color_.greenF(), color_.blueF(), 0.5);
+    glLineWidth(3);
+    QColor base_color = color_;
+    if (laps_.size() != 0)
+    {
+      for (int i = 0; i < laps_.size(); i++)
+      {
+        int hue = color_.hue() + (i + 1) * 10 * 3.1415926;
+        if (hue > 360)
+        {
+          hue %= 360;
+        }
+
+        int sat = color_.saturation();
+        int v = color_.value();
+        base_color.setHsv(hue, sat, v);
+        glColor4f(base_color.redF(), base_color.greenF(), base_color.blueF(),
+                  0.5);
+
+        if (draw_style_ == LINES)
+        {
+          glLineWidth(3);
+          glBegin(GL_LINE_STRIP);
+        }
+        else
+        {
+          glPointSize(6);
+          glBegin(GL_POINTS);
+        }
+
+        std::list<StampedPoint>::iterator it = laps_[i].begin();
+        for (; it != laps_[i].end(); it++)
+        {
+          if (it->transformed)
+          {
+            glVertex2f(it->transformed_point.getX(),
+                       it->transformed_point.getY());
+          }
+        }
+        glEnd();
+      }
+    }
+
+    if (draw_style_ == LINES)
+    {
+      glLineWidth(3);
+      glBegin(GL_LINE_STRIP);
+    }
+    else
+    {
+      glPointSize(6);
+      glBegin(GL_POINTS);
+    }
+
+    glColor4f(base_color.redF(), base_color.greenF(), base_color.blueF(), 0.5);
+
+    if (points_.size() > 0)
+    {
+      std::list<StampedPoint>::iterator it = points_.begin();
+      for (; it != points_.end(); ++it)
+      {
+        if (it->transformed)
+        {
+          glVertex2f(it->transformed_point.getX(),
+                     it->transformed_point.getY());
+          transformed = true;
+        }
+      }
+    }
+
+    glEnd();
+    return transformed;
+  }
+
+  bool PointDrawingPlugin::DrawLapsArrows()
+  {
+    bool transformed = false;
+    glColor4f(color_.redF(), color_.greenF(), color_.blueF(), 0.5);
+    glLineWidth(2);
+    QColor base_color = color_;
+    if (laps_.size() != 0)
+    {
+      for (int i = 0; i < laps_.size(); i++)
+      {
+        int hue = color_.hue() + (i + 1) * 10 * 3.1415926;
```

Comment:
See earlier comment about magic numbers. If this is a common operation, it should be extracted into a function with an explanatory name.